homeassistant/components/nest/config_flow.py

```diff
@@ -111,7 +111,7 @@ async def async_step_user(self, user_input=None):
     async def async_step_init(self, user_input=None):
         """Handle a flow start."""
         if self.is_sdm_api():
-            return None
+            raise InvalidStateError("Step only supported for legacy API")
```

Comment:
I'm not sure if we should use this exception. Until now it is only used by the core StateMachine. I'd advise to define a new exception in this module that inherits HomeAssistantError to raise.